src/core/server/plugins/plugins_service.ts

```diff
@@ -228,16 +229,11 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
           const configDescriptor = this.pluginConfigDescriptors.get(pluginId)!;
           return [
             pluginId,
-            this.configService.atPath(plugin.configPath).pipe(
-              map((config: any) =>
-                pick(
-                  config || {},
-                  Object.entries(configDescriptor.exposeToBrowser!)
-                    .filter(([_, exposed]) => exposed)
-                    .map(([key, _]) => key)
-                )
-              )
-            ),
+            this.configService
```

Comment:
Mind the `filter` in line 225 (I couldn't comment there): since the root values do not necessarily are `boolean` now, are we OK with validating that a key simply exists?

Reply:
We're validating that a value exists, but that doesn't change much. Yea, I think we're fine.

Worse would be a scenario where the descriptor only defines nested keys with false values:

```ts
{
   nested: {
      key: false,
   }
}
```

this would 'bypass' the filtering. But the output of `createBrowserConfig` would then be an empty object regardless, so I think it's fine.

TBH we could probably get rid of this whole `filter` block and let `createBrowserConfig` generates empty config for all plugins, but I wanted to avoid touching more code than necessary here.